packages/indexer/indexer/chain.py

```diff
@@ -30,32 +40,73 @@ async def is_valid_response(self, resp: aiohttp.ClientResponse) -> bool:
     async def get_json(self, resp: aiohttp.ClientResponse) -> dict:
         return json.loads(await resp.read())
 
+    def get_next_api(self):
+        return list(self.apis.keys())[self.current_api_index]
+
+    def add_api_miss(self, api: str):
```

Comment:
Looks fine but I thought we discussed storing a dict so you won't have issues with the index.